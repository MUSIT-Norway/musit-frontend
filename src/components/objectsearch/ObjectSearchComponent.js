import React from 'react'
import { I18n } from 'react-i18nify'
import { Grid, Form, FormGroup, FormControl, ControlLabel, Button, Table } from 'react-bootstrap'
import FontAwesome from 'react-fontawesome'
import Breadcrumb from '../../layout/Breadcrumb'
import { createBreadcrumbPath } from '../../util'
import PagingToolbar from '../../util/paging'

export function renderParam(id, props, style) {
  return (
    <FormGroup controlId={id}>
      <ControlLabel>{I18n.t(`musit.objectsearch.${id}.label`)}</ControlLabel>
      {' '}
      <FormControl
        style={{ ...style }}
        type="text"
        placeholder={I18n.t(`musit.objectsearch.${id}.placeHolder`)}
        value={props.params[id] || ''}
        onChange={(e) => props.onChangeField(id, e.target.value)}
      />
    </FormGroup>
  )
}

export function renderBreadcrumb(nodes: []) {
  return <Breadcrumb nodes={nodes} allActive />
}

export default (props) =>
  <div style={{ paddingTop: 20 }}>
    <main>
      <Grid>
        <div>
          <h2>{I18n.t('musit.objectsearch.title')}</h2>
          <Form inline>
            {renderParam('museumNo', props)}
            {' '}
            {renderParam('subNo', props)}
            {' '}
            {renderParam('term', props, { width: '470px' })}
            {' '}
            <Button
              type="submit"
              onClick={(e) => {
                e.preventDefault()
                props.searchForObjects(props.params)
              }}
            >
              <FontAwesome name="search" style={{ fontSize: '1.3em' }} />
            </Button>
          </Form>
          <br />
          <h4>
          {props.loaded &&
            (props.data.totalMatches > 0 ?
              I18n.t('musit.objectsearch.results.title', { count: props.data.totalMatches })
              :
              I18n.t('musit.objectsearch.results.noHit')
            )
          }
          </h4>
          {props.data.matches.length > 0 &&
            <div>
              <PagingToolbar
                numItems={props.data.totalMatches}
                baseUrl={props.location.pathname}
                currentPage={props.params.currentPage}
                perPage={props.params.perPage}
                onClick={(page) => props.searchForObjects(props.params, page)}
              />
              <Table>
                <thead>
                  <tr>
                    <th>{I18n.t('musit.objectsearch.museumNo.label')}</th>
                    <th>{I18n.t('musit.objectsearch.subNo.label')}</th>
                    <th>{I18n.t('musit.objectsearch.term.label')}</th>
                    <th>{I18n.t('musit.objectsearch.location.label')}</th>
                  </tr>
                </thead>
                <tbody>
                {props.data.matches.map((data, i) => {
                  const path = createBreadcrumbPath(data.path, data.pathNames)
                  return (
                    <tr key={i}>
                      <td className="museumNo">{data.museumNo}</td>
                      <td className="subNo">{data.subNo}</td>
                      <td className="term">{data.term}</td>
                      <td className="path">{renderBreadcrumb(path)}</td>
                      <td className="move">
                        <a
                          href=""
                          onClick={(e) => {
                            e.preventDefault()
                            props.pickObject(data, path)
                          }}
                          title={I18n.t('musit.objectsearch.addToPickList')}
                        >
                          <FontAwesome name="shopping-cart" style={{ fontSize: '1.3em' }} />
                        </a>
                      </td>
                    </tr>
                  )
                })}
                </tbody>
              </Table>
              <PagingToolbar
                numItems={props.data.totalMatches}
                baseUrl={props.location.pathname}
                currentPage={props.params.currentPage}
                perPage={props.params.perPage}
                onClick={(page) => props.searchForObjects(props.params, page)}
              />
            </div>
          }
        </div>
      </Grid>
    </main>
  </div>