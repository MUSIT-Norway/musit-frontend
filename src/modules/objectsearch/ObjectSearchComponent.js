import React from 'react';
import {I18n} from 'react-i18nify';
import {Grid, Form, FormGroup, FormControl, ControlLabel, Button, Table} from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import Breadcrumb from '../../components/layout/Breadcrumb';
import PagingToolbar from '../../shared/paging';
import {hashHistory} from 'react-router';
import Loader from 'react-loader';


export function renderParam(id, props, style) {
  return (
    <FormGroup controlId={id}>
      <ControlLabel>{I18n.t(`musit.objectsearch.${id}.label`)}</ControlLabel>
      {' '}
      <FormControl
        style={{...style}}
        type="text"
        placeholder={I18n.t(`musit.objectsearch.${id}.placeHolder`)}
        value={props.params[id] || ''}
        onChange={(e) => props.onChangeField(id, e.target.value)}
      />
    </FormGroup>
  );
}

export default (props) =>
  <div style={{paddingTop: 20}}>
    <main>
      <Grid>
        <div>
          <h1>{I18n.t('musit.objectsearch.title')}</h1>
          <Form inline>
            {renderParam('museumNo', props)}
            {' '}
            {renderParam('subNo', props)}
            {' '}
            {renderParam('term', props, {width: '440px'})}
            {' '}
            <Button
              type="submit"
              onClick={(e) => {
                e.preventDefault();
                props.searchForObjects(props.params, 1, props.user.museumId, props.user.collectionId);
              }}
            >
              <FontAwesome name="search" style={{fontSize: '1.3em'}}/>
            </Button>
          </Form>
          <br />
          <h4>
            {props.loaded &&
            (props.data.totalMatches > 0 ?
                I18n.t('musit.objectsearch.results.title', {count: props.data.totalMatches})
                :
                I18n.t('musit.objectsearch.results.noHit')
            )
            }
          </h4>
          <Loader loaded={!props.loading}>
          {props.data.matches.length > 0 &&
          <div>
            <PagingToolbar
              numItems={props.data.totalMatches}
              baseUrl={props.location.pathname}
              currentPage={props.params.currentPage}
              perPage={props.params.perPage}
              onClick={(page) => props.searchForObjects(props.params, page, props.user.museumId, props.user.collectionId)}
            />
            <Table>
              <thead>
              <tr>
                <th>{I18n.t('musit.objectsearch.museumNo.label')}</th>
                <th>{I18n.t('musit.objectsearch.subNo.label')}</th>
                <th>{I18n.t('musit.objectsearch.term.label')}</th>
                <th>{I18n.t('musit.objectsearch.location.label')}</th>
                <th>
                  <a
                    href=""
                    onClick={(e) => {
                      e.preventDefault();
                      props.pickObjects(props.data.matches, props.user.museumId, props.user.collectionId);
                    }}
                    title={I18n.t('musit.objectsearch.addAllToPickList')}
                  >
                    <FontAwesome name="shopping-cart" style={{fontSize: '1.3em'}}/>
                  </a>
                </th>
              </tr>
              </thead>
              <tbody>
                {props.data.matches.map((data, i) => {
                  const isMainObject = !data.mainObjectId || data.isMainObject();
                  const isChildObject = data.mainObjectId && !data.isMainObject();
                  return (
                    <tr key={i} className={isChildObject ? 'childObject' : isMainObject && 'mainObject'}>
                      <td className="museumNo">{data.museumNo}</td>
                      <td className="subNo">{data.subNo}</td>
                      <td className="term">{data.term}</td>
                      <td className="path">
                        {data.breadcrumb.length > 0 &&
                        <Breadcrumb
                          node={data}
                          allActive
                          onClickCrumb={(node) => hashHistory.push(node.url) }
                        />
                        }
                      </td>
                      <td className="move">
                        {isMainObject &&
                        <a
                          href=""
                          onClick={(e) => {
                            e.preventDefault();
                            props.pickObject(data, data.breadcrumb, props.user.museumId, props.user.collectionId);
                          }}
                          title={I18n.t('musit.objectsearch.addToPickList')}
                        >
                          <FontAwesome name="shopping-cart" style={{fontSize: '1.3em'}}/>
                        </a>
                        }
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            </Table>
            <PagingToolbar
              numItems={props.data.totalMatches}
              currentPage={props.params.currentPage}
              perPage={props.params.perPage}
              onClick={(page) => props.searchForObjects(props.params, page, props.user.museumId, props.user.collectionId)}
            />
          </div>
          }
          </Loader>
        </div>
      </Grid>
    </main>
  </div>;