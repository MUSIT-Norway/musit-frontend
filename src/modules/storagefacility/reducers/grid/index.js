import Config from '../../../../config';
import { mapToFrontend } from '../mapper';
import { apiUrl } from '../../../../util';

export const LOAD_SEVERAL = 'musit/storageunit-grid/LOAD_SEVERAL';
export const LOAD_SEVERAL_SUCCESS = 'musit/storageunit-grid/LOAD_SEVERAL_SUCCESS';
export const LOAD_SEVERAL_FAIL = 'musit/storageunit-grid/LOAD_SEVERAL_FAIL';
export const LOAD_ONE = 'musit/storageunit-grid/LOAD_ONE';
export const LOAD_ONE_SUCCESS = 'musit/storageunit-grid/LOAD_ONE_SUCCESS';
export const LOAD_ONE_FAIL = 'musit/storageunit-grid/LOAD_ONE_FAIL';
export const CLEAR_ROOT = 'musit/storageunit-grid/CLEAR_ROOT';
export const DELETE = 'musit/storageunit-grid/DELETE';
export const DELETE_SUCCESS = 'musit/storageunit-grid/DELETE_SUCCESS';
export const DELETE_FAIL = 'musit/storageunit-grid/DELETE_FAIL';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: mapToFrontend(action.result)
      }
    };
  case LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case DELETE:
    return {
      ...state,
      loading: true
    };
  case DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;

export const loadRoot = (id, museumId, currentPage, callback) => {
  let action = {};
  if (id) {
    action = {
      types: [LOAD_ONE, LOAD_ONE_SUCCESS, LOAD_ONE_FAIL],
      promise: (client) => client.get(apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}/${id}`)),
      callback
    };
  } else {
    action = {
      types: [LOAD_SEVERAL, LOAD_SEVERAL_SUCCESS, LOAD_SEVERAL_FAIL],
      promise: (client) => client.get(apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}/root`)),
      callback
    };
  }
  return action;
};

export const loadChildren = (id, museumId, currentPage, callback) => {
  const baseUrl = Config.magasin.urls.storagefacility.baseUrl(museumId);
  const url = apiUrl(`${baseUrl}/${id}/children?page=${currentPage || 1}&limit=${Config.magasin.limit}`);
  return {
    types: [LOAD_SEVERAL, LOAD_SEVERAL_SUCCESS, LOAD_SEVERAL_FAIL],
    promise: (client) => client.get(url),
    callback
  };
};

export const deleteUnit = (id, museumId, callback) => {
  return {
    types: [DELETE, DELETE_SUCCESS, DELETE_FAIL],
    promise: (client) => client.del(apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}/${id}`)),
    id,
    callback
  };
};

export const clearRoot = () => {
  return {
    type: CLEAR_ROOT
  };
};
