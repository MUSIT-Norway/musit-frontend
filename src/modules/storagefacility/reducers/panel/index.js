import Config from '../../../../config';
import { apiUrl } from '../../../../shared/util';

import { mapToFrontend, mapToBackend } from '../mapper';
export const INSERT = 'musit/storageunit-container/INSERT';
export const UPDATE = 'musit/storageunit-container/UPDATE';
export const INSERT_SUCCESS = 'musit/storageunit-container/INSERT_SUCCESS';
export const INSERT_FAIL = 'musit/storageunit-container/INSERT_FAIL';
export const LOAD = 'musit/storageunit-container/LOAD';
export const LOAD_SUCCESS = 'musit/storageunit-container/LOAD_SUCCESS';
export const LOAD_FAIL = 'musit/storageunit-container/LOAD_FAIL';

const initialState = {};

const storageUnitContainerReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case INSERT:
    return {
      ...state,
      loading: true
    };
  case INSERT_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: mapToFrontend(action.result)
    };
  case INSERT_FAIL:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.error
    };
  case UPDATE:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.data
    };
  case LOAD:
    return {
      ...state,
      loading: true
    };
  case LOAD_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: mapToFrontend(action.result)
    };
  case LOAD_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitContainerReducer;

const getUpdatedBy = (client, node, resolve) => {
  client.get(apiUrl(`${Config.magasin.urls.actor.baseUrl}/${node.updatedBy}`))
    .then(actor => resolve({
      ...node,
      updatedByName: actor.fn
    })).catch(() => resolve(node));
};

export const load = (id, museumId, callback) => {
  return {
    types: [LOAD, LOAD_SUCCESS, LOAD_FAIL],
    promise: (client) => new Promise((resolve, reject) => {
      client.get(apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}/${id}`))
        .then(node => getUpdatedBy(client, node, resolve, reject))
        .catch(error => reject(error));
    }),
    callback
  };
};

export const update = (data, museumId, callback) => {
  const dataToPost = mapToBackend(data);
  return {
    types: [INSERT, INSERT_SUCCESS, INSERT_FAIL],
    promise: (client) => new Promise((resolve, reject) => {
      client.put(apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}/${data.id}`), { data: dataToPost })
          .then(node => getUpdatedBy(client, node, resolve, reject))
          .catch(error => reject(error));
    }),
    callback
  };
};

export const insert = (parentId, museumId, data, callback) => {
  const url = apiUrl(`${Config.magasin.urls.storagefacility.baseUrl(museumId)}${!parentId ? '/root' : ''}`);
  const dataToPost = mapToBackend(data, parentId);
  return {
    types: [INSERT, INSERT_SUCCESS, INSERT_FAIL],
    promise: (client) => client.post(url, { data: dataToPost }),
    callback
  };
};
