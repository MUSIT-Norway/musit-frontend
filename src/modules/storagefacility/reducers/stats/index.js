import Config from '../../../../config';
import { apiUrl } from '../../../../util';

export const LOAD_STATS = 'musit/strageunit-stats/LOAD_STATS';
export const LOAD_STATS_SUCCESS = 'musit/strageunit-stats/LOAD_STATS_SUCCESS';
export const LOAD_STATS_FAILURE = 'musit/strageunit-stats/LOAD_STATS_FAILURE';
export const CLEAR_STATS = 'musit/strageunit-stats/CLEAR_STATS';

const initialState = {
  stats: {
    nodes: null,
    objects: null,
    totalObjects: null
  }
};

const storageUnitStatsReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case CLEAR_STATS:
    return {
      ...state,
      stats: null
    };
  case LOAD_STATS:
    return {
      ...state,
      stats: null,
      loading: true,
      loaded: false
    };
  case LOAD_STATS_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      stats: action.result
    };
  case LOAD_STATS_FAILURE:
    return {
      ...state,
      stats: null,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitStatsReducer;

export const loadStats = (id, museumId) => {
  return {
    types: [LOAD_STATS, LOAD_STATS_SUCCESS, LOAD_STATS_FAILURE],
    promise: (client) => client.get(apiUrl(`${Config.magasin.urls.thingaggregate.baseUrl(museumId)}/storagenodes/${id}/stats`))
  };
};

export const clearStats = () => {
  return {
    type: CLEAR_STATS
  };
};
