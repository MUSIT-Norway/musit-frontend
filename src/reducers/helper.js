export const getPath = (pathStr, pathNames) => {
  const nodeIds = pathStr.slice(1, -1).split(',').slice(1).map(p => parseFloat(p));
  return nodeIds.map(nodeId => {
    const pathMatch = pathNames.find(e => e.nodeId === nodeId)
    return {
      id: pathMatch.nodeId,
      name: pathMatch.name,
      url: '/magasin/' + pathMatch.nodeId
    }
  })
}
