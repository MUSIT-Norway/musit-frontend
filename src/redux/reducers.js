/*
 *  MUSIT is a museum database to archive natural and cultural history data.
 *  Copyright (C) 2016  MUSIT Norway, part of www.uio.no (University of Oslo)
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License,
 *  or any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License along
 *  with this program; if not, write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

import { routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from '../modules/app/appReducer';
import picklistReducer from '../modules/picklist/picklistReducer';
import storagePanelStateReducer from '../modules/storagefacility/reducers/panel/state';
import storagePanelReducer from '../modules/storagefacility/reducers/panel';
import storageNodeGridReducer from '../modules/storagefacility/reducers/grid/nodes';
import storageUnitModal from '../modules/storagefacility/reducers/modal';
import storageUnitStatsReducer from '../modules/storagefacility/reducers/stats';
import storageObjectGridReducer from '../modules/storagefacility/reducers/grid/objects';
import observationReducer from '../modules/observation/observationReducer';
import reportReducer from '../modules/reports/reportsReducer';
import controlReducer from '../modules/control/controlReducer';
import observationControlGridReducer from '../modules/events/eventsReducer';
import moveHistoryReducer from '../modules/movehistory/moveHistoryReducer';
import ObjectSearchReducer from '../modules/objectsearch/reducer';
import printReducer from '../modules/print/printReducer';

const rootReducer = combineReducers({
  routing: routerReducer,
  [auth.ID]: auth.reducer,
  print: printReducer,
  picks: picklistReducer,
  storagePanelUnit: storagePanelReducer,
  storageGridUnit: storageNodeGridReducer,
  [storageUnitModal.ID]: storageUnitModal.reducer,
  storageUnitStats: storageUnitStatsReducer,
  storagePanelState: storagePanelStateReducer,
  storageObjectGrid: storageObjectGridReducer,
  observation: observationReducer,
  reports: reportReducer,
  control: controlReducer,
  observationControlGrid: observationControlGridReducer,
  movehistory: moveHistoryReducer,
  objectSearch: ObjectSearchReducer
});

export default rootReducer;
