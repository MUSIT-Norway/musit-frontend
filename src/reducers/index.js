/*
 *  MUSIT is a museum database to archive natural and cultural history data.
 *  Copyright (C) 2016  MUSIT Norway, part of www.uio.no (University of Oslo)
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License,
 *  or any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License along
 *  with this program; if not, write to the Free Software Foundation, Inc.,
 *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */

import { routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from './auth';
import picklistReducers from '../picklist/picklistReducers';
import storageReducers from '../storage/storageReducers';
import magasinReducers from '../magasin/magasinReducers';
import reportReducer from '../reports/reportsReducer';
import storageUnitModal from './storageunit/modal';
import observationReducer from './observation';
import controlReducer from '../control/controlReducer';
import observationControlGridReducer from './grid/observationcontrol';
import moveHistoryReducer from './grid/move';
import ObjectSearchReducer from './objectsearch/reducer';
import printReducer from '../print/printReducer';

const rootReducer = combineReducers({
  magasinReducers,
  routing: routerReducer,
  [auth.ID]: auth.reducer,
  print: printReducer,
  picks: picklistReducers,
  storagePanelUnit: storageReducers,
  [storageUnitModal.ID]: storageUnitModal.reducer,
  observation: observationReducer,
  reports: reportReducer,
  control: controlReducer,
  observationControlGrid: observationControlGridReducer,
  movehistory: moveHistoryReducer,
  objectSearch: ObjectSearchReducer
});

export default rootReducer;
