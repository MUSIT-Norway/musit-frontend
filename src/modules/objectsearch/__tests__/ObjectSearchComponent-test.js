import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import ObjectSearchComponent from '../ObjectSearchComponent';
import { getPath } from '../../../reducers/helper';
import MusitObject from '../../../models/object';

describe('ObjectSearchComponent', () => {
  const path = ',1,2,3,';
  const pathNames = [
    {
      nodeId: 1,
      name: 'Root',
      type: 'Organization'
    },
    {
      nodeId: 2,
      name: 'Musit',
      type: 'Building'
    },
    {
      nodeId: 3,
      name: 'Kontoret',
      type: 'Room'
    }
  ];
  const testData = {
    totalMatches: 20,
    matches: Array(20).fill(new MusitObject(
      {
        museumNo: '12345',
        subNo: '45',
        term: 'Fuglekasse',
        id: 1,
        breadcrumb: getPath({ path, pathNames }),
        path: path,
        pathNames: pathNames
      }
    ))
  };

  it('should display object 1', () => {
    const wrapper = shallow(
      <ObjectSearchComponent
        location={{
          query: {
            page: '1'
          },
          pathname: '/#/search/objects'
        }}
        loaded={true}
        data={testData}
        onChangeField={() => true}
        params={{
          museumNo: '1234',
          subNo: '455',
          term: 'Lololol',
          currentPage: 1
        }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
