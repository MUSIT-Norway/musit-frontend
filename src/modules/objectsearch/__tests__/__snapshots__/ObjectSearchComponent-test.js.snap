exports[`ObjectSearchComponent should display object 1 1`] = `1`;
